### PR TITLE
update quic-go to v0.39.1, set a static resumption token generator key

### PR DESCRIPTION
**Pull request by marten-seemann**
Blocked on the v0.39.0 quic-go release, especially https://github.com/quic-go/quic-go/pull/4066.

### CODE REVIEW

**Comment by marten-seemann** [comment]
#2582 removes the metrics tracer, which is currently causing problems here.

**Comment by marten-seemann** [comment]
Updated the PR to v0.39.1.

File: p2p/transport/quicreuse/connmgr.go [hunk 1]
```diff
@@ -25,19 +25,21 @@ type ConnManager struct {
...
 }
 
-func NewConnManager(statelessResetKey quic.StatelessResetKey, opts ...Option) (*ConnManager, error) {
+func NewConnManager(statelessResetKey quic.StatelessResetKey, tokenKey quic.TokenGeneratorKey, opts ...Option) (*ConnManager, error) {
 	cm := &ConnManager{
 		enableReuseport: true,
 		quicListeners:   make(map[string]quicListenerEntry),
 		srk:             statelessResetKey,
+		tokenKey:        tokenKey,
 	}
```
Comment (sukunrt) [inline p2p/transport/quicreuse/connmgr.go:43]: This needs to be passed to the actual quic.Transport instance, right? Like we pass stateless reset key here
https://github.com/libp2p/go-libp2p/blob/99f761127098f1a1cac564d79955cf66c1665a20/p2p/transport/quicreuse/connmgr.go#L200

```
	return &singleOwnerTransport{Transport: quic.Transport{Conn: conn, StatelessResetKey: &c.srk}, packetConn: conn}, nil

```

**Review by sukunrt — CHANGES_REQUESTED** [review]

File: p2p/transport/quicreuse/connmgr.go (hunk 1 above)
Comment (marten-seemann) [reply p2p/transport/quicreuse/connmgr.go:43]: Good catch! Fixed in 3a6d0ee74fbf168df70da4ba6491a1ed910e6090.

**Comment by marten-seemann** [comment]
Rebasing in the hope that this helps with flaky tests that @sukunrt recently fixed.

File: p2p/transport/quicreuse/connmgr.go [hunk 2]
```diff
@@ -25,19 +25,21 @@ type ConnManager struct {
...
 	ln       *quicListener
 }
 
-func NewConnManager(statelessResetKey quic.StatelessResetKey, opts ...Option) (*ConnManager, error) {
+func NewConnManager(statelessResetKey quic.StatelessResetKey, tokenKey quic.TokenGeneratorKey, opts ...Option) (*ConnManager, error) {
 	cm := &ConnManager{
 		enableReuseport: true,
 		quicListeners:   make(map[string]quicListenerEntry),
 		srk:             statelessResetKey,
+		tokenKey:        tokenKey,
```
Comment (sukunrt) [inline p2p/transport/quicreuse/connmgr.go:42]: I think this needs to be passed to `reuse` struct too on lines 63-66.

**Review by sukunrt — CHANGES_REQUESTED** [review]

**Comment by sukunrt** [comment]
looks like the flaky fixes are working :)

File: p2p/transport/quicreuse/connmgr.go (hunk 2 above)
Comment (marten-seemann) [reply p2p/transport/quicreuse/connmgr.go:42]: Good catch!

**Review by sukunrt — APPROVED** [review]

**Pull request merged**